include/alpaka/kernel/TaskKernelGpuUniformCudaHipRt.hpp

```diff
@@ -390,6 +468,315 @@ namespace alpaka
+                // TApi::deviceSetLimit(TApi::limitPrintfFifoSize, printfFifoSize*10);
+                // TApi::deviceGetLimit(&printfFifoSize, TApi::limitPrintfFifoSize);
+                // std::cout << __func__ << "INFO: printfFifoSize: " << printfFifoSize << std::endl;
+#        endif
+                auto const gridBlockExtent = getWorkDiv<Grid, Blocks>(task);
+                auto const blockThreadExtent = getWorkDiv<Block, Threads>(task);
+                auto const threadElemExtent = getWorkDiv<Thread, Elems>(task);
+
+                dim3 const gridDim = uniform_cuda_hip::detail::convertVecToUniformCudaHipDim(gridBlockExtent);
+                dim3 const blockDim = uniform_cuda_hip::detail::convertVecToUniformCudaHipDim(blockThreadExtent);
+                uniform_cuda_hip::detail::checkVecOnly3Dim(threadElemExtent);
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                std::cout << __func__ << " gridDim: " << gridDim.z << " " << gridDim.y << " " << gridDim.x
+                          << " blockDim: " << blockDim.z << " " << blockDim.y << " " << blockDim.x << std::endl;
+#        endif
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+                // This checks for a valid work division that is also compliant with the maxima of the accelerator.
+                if(!isValidWorkDiv<TAcc>(getDev(queue), task))
+                {
+                    throw std::runtime_error(
+                        "The given work division is not valid or not supported by the device of type "
+                        + getAccName<AccGpuUniformCudaHipRt<TApi, TDim, TIdx>>() + "!");
+                }
+#        endif
+
+                // Get the size of the block shared dynamic memory.
+                auto const blockSharedMemDynSizeBytes = std::apply(
+                    [&](remove_restrict_t<std::decay_t<TArgs>> const&... args) {
+                        return getBlockSharedMemDynSizeBytes<TAcc>(
+                            task.m_kernelFnObj,
+                            blockThreadExtent,
+                            threadElemExtent,
+                            args...);
+                    },
+                    task.m_args);
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_FULL
+                // Log the block shared memory idx.
+                std::cout << __func__ << " BlockSharedMemDynSizeBytes: " << blockSharedMemDynSizeBytes << " B"
+                          << std::endl;
+#        endif
+                auto kernelName = alpaka::detail::
+                    gpuKernel<TKernelFnObj, TApi, TAcc, TDim, TIdx, remove_restrict_t<std::decay_t<TArgs>>...>;
+
+#        if ALPAKA_DEBUG >= ALPAKA_DEBUG_MINIMAL
+                {
+                    // This checks if requested number of blocks is compliant with the maxima of the accelerator.
+                    int numBlocksPerSm = 0;
+                    cudaOccupancyMaxActiveBlocksPerMultiprocessor(
```

Comment:
eventually, can you add an entry for this function in `include/alpaka/core/ApiCudaRt.hpp` and `include/alpaka/core/ApiHipRt.hpp` ?